README.md

```diff
@@ -76,189 +76,164 @@ await server.stop()
-  - `repoPath` string - the repository path to use for this node, ignored if node is disposable
-  - `disposable` bool (default true) - a new repo is created and initialized for each invocation, as well as cleaned up automatically once the process exits
-  - `defaultAddrs` bool (default false) - use the daemon default `Swarm` addrs
-  - `args` - array of cmd line arguments to be passed to ipfs daemon
-  - `config` - ipfs configuration options
-
-Returns a promise that resolves to:
-
-- `ipfsd` - is the daemon controller instance:
-  - `api` - a property of `ipfsd`, an instance of  [ipfs-http-client](https://github.com/ipfs/js-ipfs-http-client) attached to the newly created ipfs node
-
-**example:** See [Usage](#usage)
-
-#### Get daemon version with `f.version() : Promise`
-
-Get the version without spawning a daemon
-
-- `callback` - is a function with the signature `function(err, version)`, where version might be one of the following:
-    - if `type` is 'go' a version string like `ipfs version <version number>`
-    - if `type` is 'js' a version string like `js-ipfs version: <version number>`
-     - if `type` is 'proc' an object with the following properties:
-        - version - the ipfs version
-        - repo - the repo version
-        - commit - the commit hash for this version
-
-### Remote endpoint - `const server = IPFSFactory.createServer([options])`
-
-`IPFSFactory.createServer` starts a IPFSFactory endpoint.
+### `create([options])`
+Creates a factory.
 
-- `options` is an optional object the following properties:
-  - `port` - the port to start the server on
+- `options` **[FactoryOptions](#FactoryOptions)** Factory options.
 
-**example:**
-```js
-const IPFSFactory = require('ipfsd-ctl')
+Returns a **[Factory](#factory)**
 
-const server = IPFSFactory.createServer({ port: 12345 })
+### `createNode([options])`
+Creates a node.
 
-await server.start()
+- `options` **[FactoryOptions](#FactoryOptions)** Factory options.
 
-console.log('endpoint is running')
+Returns a **[Node](#Node)**
 
-await server.stop()
-
-console.log('endpoint has stopped')
-```
+### `createNodeTests([options])`
+Creates a node with custom configuration for tests.
 
-### IPFS Daemon Controller - `ipfsd`
+- `options` **[FactoryOptions](#FactoryOptions)** Factory options.
 
-The IPFS daemon controller (`ipfsd`) allows you to interact with the spawned IPFS daemon.
+Returns a **[Node](#Node)**
 
-#### `ipfsd.apiAddr` (getter)
+### `createTestsInterface([options])`
+Creates an interface to setup and teardown IPFS nodes for tests.
 
-Get the address (multiaddr) of connected IPFS API. Returns a multiaddr
+- `options` **[FactoryOptions](#FactoryOptions)** Factory options.
 
-#### `ipfsd.gatewayAddr` (getter)
+Returns a **[TestsInterface](#TestsInterface)**
+#### TestsInterface
+`nodes` **Array** List of the created controlled nodes.   
+`node()` **Function** Creates a standalone node, this node will **NOT** be stopped by `teardown()`.   
```

Comment:
This needs better wording

`node()` returns the a ctl node 
`setup()` returns directly an ipfs core api 

Should both return the same thing ? 

i would rather add a new param than to use an option just to keep `FactoryOptions` clean. Cause this would be specific to the testInterface.


I also think using the word `node` here is wrong... We use daemon for process nodes and node for in process etc... but here it's actually a `Controller` for a IPFS node either process or in-proc.


Would you agree if i change the naming to:
`Controller` instead of `Node` ?

Then ipfsd-ctl would have:

`Factory` to wrap spawn, version and tmpDir
`Controller` to wrap the 3 different types of node
plus some methods that return an IPFS Core API.